Apps/Admin/Client/Store/BlockAccess/BlockAccessEffects.cs

```diff
@@ -0,0 +1,88 @@
+//  limitations under the License.
+// -------------------------------------------------------------------------
+namespace HealthGateway.Admin.Client.Store.BlockAccess
+{
+    using System;
+    using System.Net.Http;
+    using System.Threading.Tasks;
+    using Fluxor;
+    using HealthGateway.Admin.Client.Api;
+    using HealthGateway.Admin.Client.Store.PatientDetails;
+    using HealthGateway.Admin.Client.Store.PatientSupport;
+    using HealthGateway.Admin.Client.Utils;
+    using HealthGateway.Admin.Common.Models;
+    using Microsoft.Extensions.Logging;
+    using Refit;
+
+#pragma warning disable CS1591, SA1600
+    public class BlockAccessEffects
+    {
+        public BlockAccessEffects(ISupportApi api, ILogger<BlockAccessEffects> logger, IState<PatientSupportState> patientSupportState)
+        {
+            this.Api = api;
+            this.Logger = logger;
+            this.PatientSupportState = patientSupportState;
+        }
+
+        private ISupportApi Api { get; }
+
+        private ILogger<BlockAccessEffects> Logger { get; }
+
+        private IState<PatientSupportState> PatientSupportState { get; }
+
+        [EffectMethod]
+        public async Task HandleSetBlockAccessAction(BlockAccessActions.SetBlockAccessAction action, IDispatcher dispatcher)
+        {
+            this.Logger.LogInformation("Setting block access");
+            try
+            {
+                string? patientHdid = this.PatientSupportState.Value.Result?[0].Hdid;
+                if (patientHdid == null)
+                {
+                    RequestError error = new() { Message = "No patient selected" };
+                    dispatcher.Dispatch(new BlockAccessActions.SetBlockAccessFailureAction(error));
+                    return;
+                }
+
+                BlockAccessRequest request = new(action.DataSources, action.Reason);
+                await this.Api.BlockAccessAsync(patientHdid, request).ConfigureAwait(true);
+                dispatcher.Dispatch(new BlockAccessActions.SetBlockAccessSuccessAction());
+            }
+            catch (Exception e) when (e is ApiException or HttpRequestException)
+            {
+                this.Logger.LogError("Error setting block access: {Exception}", e.ToString());
+                RequestError error = StoreUtility.FormatRequestError(e);
+                dispatcher.Dispatch(new BlockAccessActions.SetBlockAccessFailureAction(error));
+            }
+        }
+
+        [EffectMethod(typeof(BlockAccessActions.SetBlockAccessSuccessAction))]
+        public Task HandleSetBlockSuccessAction(IDispatcher dispatcher)
+        {
+            string? patientHdid = this.PatientSupportState.Value.Result?[0].Hdid;
+            if (patientHdid == null)
+            {
+                RequestError error = new() { Message = "No patient selected" };
+                dispatcher.Dispatch(new BlockAccessActions.SetBlockAccessFailureAction(error));
+                return Task.CompletedTask;
+            }
```

Comment:
As above, the HDID should instead be passed as a property of the action.

Reply:
This isn't the way it was done in the delegation store, it would retrieve the delegate from the state, but its not guaranteed that we clicked on the first patient, I'll rework this.